packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js

```diff
@@ -33,34 +31,29 @@ const { prefix } = settings;
  */
 const TableOfContents = ({ menuItems, children, menuLabel, theme }) => {
   const [selectedId, setSelectedId] = useState(menuItems[0].id);
-  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].title);
-
+  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].name);
   useEffect(() => {
+    if (!menuItems || menuItems === undefined || menuItems === null) {
+      findMenuItems();
```

Comment:
@carthick one thing I'm not a fan about here is that you're mutating the prop `menuItems` coming into the component. If anything, you might want to do something like:
```
const useMenuItems = menuItems ? [...menuItems] : _findMenuItems();
```

In this case, you check if `menuItems` is defined or not, and if not, it calls the `_findMenuItems` method (which I still think should be taken completely outside of the class), and use the return of that to set `useMenuItems` within the class. If  `menuItems` is defined, you're using the spread operator to do a safe clone of the array instead of manipulating `menuItems` within the class. After that, I'd swap out usage of `menuItems` in here with `useMenuItems`.

Reply:
@jeffchew as pre my understanding i have updated the .. Please let me know if i need to update anything else....